File: doctor/src/main/java/com/sumian/sddoctor/widget/SumianSwipeRefreshLayout.java
package com.sumian.sddoctor.widget;

import android.content.Context;
import android.util.AttributeSet;

import com.sumian.sddoctor.R;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

/**
 * Created by jzz
 * on 2018/2/1.
 * desc:
 */

public class SumianSwipeRefreshLayout extends SwipeRefreshLayout {

    private static final long DELAY_MILLS = 6 * 1000L;
    private Runnable mDismissRunnable = this::hideRefreshAnim;

    public SumianSwipeRefreshLayout(@NonNull Context context) {
        this(context, null);
    }

    public SumianSwipeRefreshLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        initView();
    }

    private void initView() {
        this.setColorSchemeResources(R.color.b3_color, R.color.b7_color);
    }

    public void showRefreshAnim() {
        setRefreshing(true);
        removeCallbacks(mDismissRunnable);
        postDelayed(mDismissRunnable, DELAY_MILLS);
    }

    public void hideRefreshAnim() {
        removeCallbacks(mDismissRunnable);
        setRefreshing(false);
    }
}

File: common/src/main/java/com/sumian/common/widget/refresh/SumianSwipeRefreshLayout.kt
package com.sumian.common.widget.refresh

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.sumian.common.R


/**
 * Created by jzz
 * on 2018/2/1.
 * desc:
 */

@Suppress("DEPRECATION")
open class SumianSwipeRefreshLayout @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) : SwipeRefreshLayout(context, attrs) {

    private val mDismissRunnable = Runnable { this.hideRefreshAnim() }

    init {
        initView(attrs)
    }

    private fun initView(attrs: AttributeSet?) {
        val attributes = context.obtainStyledAttributes(attrs, R.styleable.SumianSwipeRefreshLayout)
        val progressColor = attributes.getColor(R.styleable.SumianSwipeRefreshLayout_ssrl_progress_color, resources.getColor(R.color.b3_color))
        val progressBgColor = attributes.getColor(R.styleable.SumianSwipeRefreshLayout_ssrl_progress_bg_color, resources.getColor(R.color.b2_color))
        attributes.recycle()
        setColorSchemeColors(progressColor)
        setProgressBackgroundColorSchemeColor(progressBgColor)
    }

    fun showRefreshAnim() {
        isRefreshing = true
        removeCallbacks(mDismissRunnable)
        postDelayed(mDismissRunnable, DELAY_MILLS)
    }

    fun hideRefreshAnim() {
        removeCallbacks(mDismissRunnable)
        isRefreshing = false
    }

    companion object {

        private const val DELAY_MILLS = 6 * 1000L
    }
}
